assignment/assignment.hs

```diff
@@ -17,6 +17,8 @@ import Data.Char(intToDigit)
 -- Implement a version vcalled `myreverse` using `foldl`.
 
 -- myreverse :: [a] -> [a]
+switch x y = y:x
+myreverse a = foldl (switch) [] a
 
```

Comment:
Nice.

My only comment would be that Haskell folk would probably remove the `a` from each side:

~~~ hs
myreverse = foldl (switch) []
~~~

They call that process `eta reduction` in that smug way they have :)